src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -458,43 +430,20 @@ public WasmFetchResponse(JSObject fetchResponse, JSObject abortController, Cance
 
         public void ThrowIfDisposed()
         {
-#if FEATURE_WASM_THREADS
-            lock (ThisLock)
+            lock (this)
```

Comment:
iirc it's bad lock hygiene to ever do `lock (this)`, but since this is an internal class it's not a big deal. maybe just worth being careful about

Reply:
because it's internal it was the reason why I didn't want to allocate extra object.